main.js

```diff
@@ -1,24 +1,20 @@
 function appendRepeat(Promise) {
-  Promise.repeat = function(callback) {
-    return new Promise((resolve, reject) => {
-			let iteration = 0;
-	  	const responses = [];
-      const recurse = () => {
-        callback(iteration++)
-          .then(response => {
-            if(response) {
-			  			responses.push(response);
-              return recurse(callback);
-            }
-        		resolve(responses); 
-          })
-          .catch(err => reject(err));
-      };
-      return recurse();
-    });
+  Promise.repeat = function(promise) {
+		let iteration = 0;
+		let chainedPromise = promise(iteration++);
+		let results = [];
+
+		let handleIteration = result => {
+			if (result == undefined) return results;
```

Comment:
I like, I was considering this. I'll have to add this change to version 2.0.0. It is a breaking change to the API, but I think it's best.